packages/nodejs/src/__tests__/transmitter.test.ts

```diff
@@ -0,0 +1,168 @@
+import { Transmitter } from "../transmitter"
+import nock from "nock"
+import https from "https"
+import fs from "fs"
+import { ClientRequest, IncomingMessage } from "http"
+
+describe("Transmitter", () => {
+  let originalEnv: any = undefined
+
+  beforeEach(() => {
+    if (!nock.isActive()) {
+      nock.activate()
+    }
+    nock.disableNetConnect()
+
+    originalEnv = { ...process.env }
+  })
+
+  afterEach(() => {
+    jest.clearAllMocks()
+    nock.restore()
+
+    process.env = originalEnv
+  })
+
+  describe(".transmit", () => {
+    function sampleTransmitter(): Transmitter {
+      return new Transmitter("https://example.com/foo", "request body")
+    }
+
+    function mockSampleRequest({
+      query = {},
+      responseBody = { json: "response" }
+    }: any = {}) {
+      nock("https://example.com")
+        .post("/foo", "request body")
+        .query({
+          api_key: "",
+          name: "",
+          environment: "test",
+          hostname: "",
+          ...query
+        })
+        .reply(200, responseBody)
+    }
+
+    async function expectSampleResponse(
+      response: Promise<any>,
+      expectedBody: object = { json: "response" }
+    ) {
+      await expect(response).resolves.toEqual({
+        status: 200,
+        body: expectedBody
+      })
+    }
+
+    function mockHTTPSRequest() {
+      return jest.spyOn(https, "request").mockImplementation(opts => {
+        const clientRequestMock = {
+          on: () => {},
+          write: () => {},
+          end: () => {}
+        }
+
+        return (clientRequestMock as unknown) as ClientRequest
+      })
+    }
+
+    it("resolves to a response on success", async () => {
+      const transmitter = sampleTransmitter()
+
+      mockSampleRequest()
+
+      await expectSampleResponse(transmitter.transmit())
+    })
+
+    it("resolves to an empty object if the response is not JSON", async () => {
+      const transmitter = sampleTransmitter()
+
+      mockSampleRequest({ responseBody: "not JSON" })
+
+      await expectSampleResponse(transmitter.transmit(), {})
```

Comment:
Can you add a similar assertion to the test above this for a valid JSON body?

Reply:
It is implicitly done -- If you don't pass a second argument to `expectSampleResponse()` with a custom response body to expect, it expects the sample response, which is `{json: "response"}`.

Reply:
I think removing the `responseBody` default from `mockSampleRequest()` will improve the readability of the tests.

Reply:
I think so too. Now it looks like we don't assert the body on line 74.

Reply:
Agreed, went a bit too far with moving everything to default arguments. The purpose of it was to avoid having to talk about the response body at all in the "query values from config" test.

I have removed the default `{json: "response"}` object from both `mockSampleRequest` and `expectSampleResponse`, which is now just `expectResponse`, as there's nothing "sample" about it anymore. The signature of `mockSampleRequest` has been simplified as only one of its parameters is now optional.

Let me know if this is clearer, and if there's anything else that could improve it.